rasterio/_err.pyx

```diff
@@ -219,35 +230,47 @@ cdef OGRErr exc_wrap_ogrerr(OGRErr err) except -1:
         raise CPLE_BaseError(3, err, "OGR Error code {}".format(err))
 
 
+cdef class StackChecker:
+
+    def __init__(self, error_stack=None):
+        self.error_stack = error_stack or {}
+
+    cdef int exc_wrap_int(self, int err) except -1:
```

Comment:
`StackChecker` provides some familiar error checking/raising functionality, but in the context of our new Python chained error stack instead of GDAL's error stack.